DataManagementSystem/Client/FTS3Job.py

```diff
@@ -200,11 +199,19 @@ def _constructTransferJob( self, context, pinTime, allTargetSURLs, failedLFNs, t
     source_spacetoken = res['Value']
 
     # getting all the source surls
-    res = StorageElement( self.sourceSE ).getURL( allTargetSURLs, protocol = 'srm' )
+    lfns = allTargetSURLs.keys()
```

Comment:
This makes a useless copy. You could have getVOName (or its equivalent that must exist) take a list and loop over until you have the first one.